area.go

```diff
@@ -0,0 +1,258 @@
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+
+		res := &app.AreaList{}
+		res.Data = ConvertAreas(appl, ctx.RequestData, children, addResolvedPath)
+
+		return ctx.OK(res)
+	})
+}
+
+// CreateChild runs the create-child action.
+func (c *AreaController) CreateChild(ctx *app.CreateChildAreaContext) error {
+
+	_, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+	parentID, err := uuid.FromString(ctx.ID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+	}
+
+	return application.Transactional(c.db, func(appl application.Application) error {
+		parent, err := appl.Areas().Load(ctx, parentID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+		}
+
+		reqArea := ctx.Payload.Data
+		if reqArea.Attributes.Name == nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("data.attributes.name", nil).Expected("not nil"))
+		}
+
+		childPath := area.ConvertToLtreeFormat(parentID.String())
+		if parent.Path != "" {
+			childPath = parent.Path + pathSepInDatabase + childPath
+		}
+		newArea := area.Area{
+			SpaceID: parent.SpaceID,
+			Path:    childPath,
+			Name:    *reqArea.Attributes.Name,
+		}
+
+		err = appl.Areas().Create(ctx, &newArea)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+
+		res := &app.AreaSingle{
+			Data: ConvertArea(appl, ctx.RequestData, &newArea, addResolvedPath),
+		}
+		ctx.ResponseData.Header().Set("Location", rest.AbsoluteURL(ctx.RequestData, app.AreaHref(res.Data.ID)))
+		return ctx.Created(res)
+	})
+}
+
+// Show runs the show action.
+func (c *AreaController) Show(ctx *app.ShowAreaContext) error {
+	id, err := uuid.FromString(ctx.ID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+	}
+
+	return application.Transactional(c.db, func(appl application.Application) error {
+		a, err := appl.Areas().Load(ctx, id)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+
+		res := &app.AreaSingle{}
+		res.Data = ConvertArea(appl, ctx.RequestData, a, addResolvedPath)
+		return ctx.OK(res)
+	})
+}
+
+// addResolvedPath resolves the path in the form of /area1/area2/area3
+func addResolvedPath(appl application.Application, req *goa.RequestData, mArea *area.Area, sArea *app.Area) error {
+	pathResolved, error := getResolvePath(appl, mArea)
+	sArea.Attributes.ParentPathResolved = pathResolved
+	return error
+
+}
+
+func getResolvePath(appl application.Application, a *area.Area) (*string, error) {
+	parentUuidStrings := strings.Split(area.ConvertFromLtreeFormat(a.Path), pathSepInService)
+	parentUuids := convertToUuid(parentUuidStrings)
+	parentAreas, err := appl.Areas().LoadMultiple(context.Background(), parentUuids)
+	if err != nil {
+		return nil, err
+	}
+	pathResolved := ""
+	for _, a := range parentAreas {
+		if pathResolved == "" {
```

Comment:
Thanks for the feedback @pranavgore09 , Good point, there is no guarantee about the order, I will handle the ordering in the code.